.github/workflows/update-readme-table.yaml

```diff
@@ -0,0 +1,85 @@
+        python libraries-bom-table-generation/scripts/parse_pom.py
+    - name: Update repo_versions
+      run: |
+        echo "Updating repo versions for checkout and handwritten modules"
+        python libraries-bom-table-generation/scripts/update_repo_versions.py
+        git checkout -
+    - name: Update for new modules and links for sdk-platform-java and google-cloud-java
+      run: |
+        while read -r repo directory tag artifact
+        do
+          echo "REPO=$repo"
+          echo "DIRECTORY=$directory"
+          echo "TAG=$tag"
+          git clone https://github.com/${repo}.git
+          git -C "${directory}" checkout $tag
+          echo "Updating javaModulesVersions.yaml file using ${repo}'s versions.txt file"
+          python libraries-bom-table-generation/scripts/parse_versionstxt.py ${directory}/versions.txt
+          if [[ "$repo" == "googleapis/google-cloud-java" ]]; then
+            echo "Updating javaModulesLibraryReferenceLinks.yaml, javaModulesNamePretty.yaml, and javaModulesProductReferenceLinks.yaml files using ${repo}'s client/product links"
+            python libraries-bom-table-generation/scripts/updateClientProductDocumentationLinks.py # Update client/product doc links based on .repo-metadata.json files
+            echo "Updating google-cloud-java_javadocs_modules.txt"
+            python libraries-bom-table-generation/scripts/update_googlecloudjava_javadocs_modules.py ${directory}/versions.txt # Update for any new modules in google-cloud-java
+          fi
+          if [[ "$repo" == "googleapis/sdk-platform-java" ]]; then
+            echo "Updating javaModulesVersions.yaml file for the auth library"
+            python libraries-bom-table-generation/scripts/parse_pom_auth_library.py ${directory}/gapic-generator-java-pom-parent/pom.xml # Update auth library version
+          fi
+        done < libraries-bom-table-generation/helpers/repos_for_versionstxt.txt
+    - name: Update links for handwritten libraries
+      run: |
+        while read -r repo directory tag artifact
+        do
+          echo "REPO=$repo"
+          echo "DIRECTORY=$directory"
+          echo "TAG=$tag"
+          git clone https://github.com/${repo}.git
+          git -C "${directory}" checkout $tag
+          echo "Updating client and product documentation links using ${repo}'s .repo-metadata.json file"
+          python libraries-bom-table-generation/scripts/updateClientProductDocumentationLinks.py # Update client/product doc links based on .repo-metadata.json files
+        done < libraries-bom-table-generation/helpers/handwritten_libraries_javadocs_modules.txt
+    - name: Update libraryTable
+      run: |
+        echo "Updating libraryTable"
+        python libraries-bom-table-generation/scripts/libraryTable_generation.py
+    - name: Update README
+      run: |
+        echo "Updating README"
+        python libraries-bom-table-generation/scripts/updateREADME.py
+
```

Comment:
Would it make sense to combine all of these steps into just one? This is a lot of code in a workflow file that is hard to run locally. Let's try to couple ourselves less to GitHub Actions if possible and allow easier local testing.

Reply:
> Would it make sense to combine all of these steps into just one? This is a lot of code in a workflow file that is hard to run locally. Let's try to couple ourselves less to GitHub Actions if possible and allow easier local testing.

I just discovered that you can view raw files in other repos by tag (e.g. https://raw.githubusercontent.com/googleapis/sdk-platform-java/v2.23.0/versions.txt). Let me see if I can create a python script that would replace these Github Actions of checking out various repos and instead downloads the files we need instead of having to clone the whole repo.

Reply:
Have refactored this to just run a single python script and then commit the updated README file